controllers/flashsystemcluster_controller.go

```diff
@@ -350,6 +330,30 @@ func (r *FlashSystemClusterReconciler) reconcile(instance *odfv1alpha1.FlashSyst
 	return reconcile.Result{}, nil
 }
 
+func (r *FlashSystemClusterReconciler) ensureSecretOwnership(instance *odfv1alpha1.FlashSystemCluster, newOwnerForSecret v1.OwnerReference) error {
+	// Reading the secret, if it has ownership then skip, else update the secret with details of the FlashSystemCluster
+	secret := &corev1.Secret{}
+	err := r.Client.Get(
+		context.TODO(),
+		types.NamespacedName{Name: instance.Spec.Secret.Name, Namespace: instance.Spec.Secret.Namespace},
+		secret)
+
+	if err != nil {
+		r.Log.Error(err, "flashSystemCluster secret not found")
```

Comment:
The error could be different than 'not found' error

Reply:
changed to -
r.Log.Error(err, "failed to get secret for FlashSystemCluster")